lib/Conversion/HWArithToHW/HWArithToHW.cpp

```diff
@@ -222,8 +259,11 @@ struct ICmpOpLowering : public OpConversionPattern<ICmpOp> {
     Value rhsValue = extendTypeWidth(rewriter, loc, adaptor.getRhs(), cmpWidth,
                                      rhsType.isSigned());
 
-    rewriter.replaceOpWithNewOp<comb::ICmpOp>(op, combPred, lhsValue, rhsValue,
-                                              false);
+    auto newOp = rewriter.replaceOpWithNewOp<comb::ICmpOp>(
+        op, combPred, lhsValue, rhsValue, false);
+    auto predAttr = newOp.getPredicateAttr();
+    newOp->setAttrs(op->getAttrs());
+    newOp->setAttr(newOp.getPredicateAttrName(), predAttr);
```

Comment:
Not sure if I am a fan of this approach. If the operations get new internal attributes (or get renamed), these would be propagated too, wouldn't it?
Is there a particular reason why you didn't choose an explicit attribute propagation as in: https://github.com/llvm/circt/blob/b5e6fa769a580faa1c5ebda29d8a01c45633bc08/lib/Dialect/Comb/CombFolds.cpp#L61-L64

Reply:
I agree. It's reasonable (and relatively easy) to just copy the dialect attributes. See `op.(get|set)DialectAttrs`.

Reply:
It would also avoid having to re-set the pred attr.

Reply:
Agree on all points - will change to get/setDialectAttrs.